src/bvh/iter.rs

```diff
@@ -33,7 +33,14 @@ impl<'bvh, 'shape, T: BHValue, const D: usize, Shape: Bounded<T, D>>
             stack: [0; 32],
             node_index: 0,
             stack_size: 0,
-            has_node: !bvh.nodes.is_empty(),
+            has_node: match bvh.nodes.first() {
+                // Only process the root leaf node if the shape's AABB is intersected.
```

Comment:
Maybe provide some reasoning here _why_ this is a good idea (even if it seems obvious to you).

Reply:
Since this logic was duplicated, and I didn't want to also duplicate the comment, I abstracted this into a function.